src/client/datascience/jupyter/kernels/helpers.ts

```diff
@@ -576,13 +603,43 @@ export function findPreferredKernel(
                         metadata.interpreter?.path?.toLowerCase().includes('python3') ||
                         spec.argv[0].toLocaleLowerCase().includes('python3'))
                 ) {
+                    traceInfoIf(isCI, 'Increased score by +1 for matching major version 3');
                     score += 1;
                     subScore += 1;
                     traceInfo(
                         `findPreferredKernel score for Python3, ${getDisplayNameOrNameOfKernelConnection(
                             metadata
                         )} is ${score}`
                     );
+
+                    // If the versions match exactly, then increase the score by another 1
+                    if (
+                        typeof notebookMetadata === 'object' &&
+                        'interpreter' in notebookMetadata &&
+                        notebookMetadata.interpreter &&
+                        typeof notebookMetadata.interpreter === 'object' &&
+                        'hash' in notebookMetadata.interpreter &&
+                        metadata.kind === 'startUsingPythonInterpreter' &&
+                        notebookMetadata.interpreter
+                    ) {
+                        const nbMetadataInterpreter = notebookMetadata.interpreter as Partial<PythonEnvironment>;
+                        if (
+                            nbMetadataInterpreter.version?.raw &&
+                            nbMetadataInterpreter.version?.raw === metadata.interpreter.version?.raw
+                        ) {
+                            traceInfoIf(
+                                isCI,
+                                'Increased score by +3 for matching raw version in notebook metadata interpreter'
+                            );
+                            score += 3;
+                        }
+                    } else if (
+                        metadata.interpreter?.version?.raw &&
+                        metadata.interpreter?.version?.raw === preferredInterpreter?.version?.raw
+                    ) {
+                        traceInfoIf(isCI, 'Increased score by +3 for matching raw version in preferred interpreter');
```

Comment:
Cam across a scenario where 3.8.10 was picked over 3.8.11 (beacuse both have 3 and 3.8.10 was on the top of the list hence that got picked when both had the same score).